[IMP] sql_export_mail: allow partners as email recipients

DjodyKort opened the pull request:
## What does this PR do?
Add a **backported feature** for sending SQL export reports to partners in addition to users. This enables more flexible notification workflows by allowing partners to receive automated reports via email.

## Migration Notes
This feature is backported from v18/16 to v17 so it already existed.

## Testing
All pre-commit hooks passed including:
- ✅ Ruff (Python linting)
- ✅ Pylint-odoo (mandatory checks)
- ✅ OCA module structure checks
- ✅ XML validation

Unit tests:
- ✅ test_sql_query_mail_partner (new)
- ✅ All existing tests still pass

Manual testing performed with partner email notifications.

Larsq1 commented on sql_export_mail/__manifest__.py:
```diff
@@ -1,7 +1,7 @@
 {
     "name": "SQL Export Mail",
-    "version": "17.0.1.0.0",
+    "version": "17.0.1.0.2",
```
This can be 17.0.1.1.0

DjodyKort replied:
Alright

legalsylvain replied:
please, dont change this key in the manifest. it will be done, when merging.

Larsq1 reviewed (approved):
LGTM

Larsq1 commented on sql_export_mail/models/sql_export.py:
```diff
@@ -123,7 +131,7 @@ def check_no_parameter_if_sent_by_mail(self):
             if export.query_properties_definition and export.mail_user_ids:
                 raise UserError(
-                    _(
+                    self.env._(
```
self.env_() is an Odoo 18 feature, this does not work in v17

DjodyKort replied:
Noted

Larsq1 commented on sql_export_mail/models/sql_export.py:
```diff
@@ -134,15 +142,47 @@ def check_mail_user(self):
                 if not user.email:
-                    raise UserError(_("The user does not have any e-mail address."))
+                    raise UserError(
+                        self.env._("The user does not have any e-mail address.")
```
```suggestion
                        _("The user does not have any e-mail address.")
```

Larsq1 replied:
and in all other places

Larsq1 reviewed (changes requested):
See comments

Commit:
[IMP] sql_export_mail: allow partners as email recipients

Backported from Odoo 18 version into 17.0 branch.

- Add mail_partner_ids field to send SQL export reports to partners

- Include partner emails in mail template alongside user emails
